library/src-3.x/scala/quoted/Expr.scala

```diff
@@ -2,14 +2,16 @@ package scala
 
 package quoted {
 
+  import scala.quoted.show.SyntaxHighlight
+
   sealed abstract class Expr[+T] {
 
     /** Evaluate the contents of this expression and return the result.
      *
      *  May throw a FreeVariableError on expressions that came from a macro.
      */
     @deprecated("Use scala.quoted.run", "")
-    final def run(implicit toolbox: Toolbox): T = toolbox.run(this)
+    final def run(implicit toolbox: Toolbox): T = toolbox.run(_ => this)
```

Comment:
Thanks to the explanation by @nicolasstucki, now I understand `run` is double-meaning for a compiled language: it means to execute the expression, but also mean inside a specific compiler run --- `Expr[T]` may only be created for a specific compiler run.

If that is the case, it seems make sense to use path-dependent types to make `Expr[T]` depend on compiler run.